src/msg/async/AsyncMessenger.cc

```diff
@@ -662,9 +660,20 @@ int AsyncMessenger::send_to(Message *m, int type, const entity_addrvec_t& addrs)
     return -EINVAL;
   }
 
-  auto av = _filter_addrs(addrs);
-  const AsyncConnectionRef& conn = _lookup_conn(av);
-  submit_message(m, conn, av, type);
+  if (cct->_conf->ms_dump_on_send) {
+    m->encode(-1, MSG_CRC_ALL);
+    ldout(cct, 0) << __func__ << " submit_message " << *m << "\n";
+    m->get_payload().hexdump(*_dout);
+    if (m->get_data().length() > 0) {
+      *_dout << " data:\n";
+      m->get_data().hexdump(*_dout);
+    }
+    *_dout << dendl;
+    m->clear_payload();
+  }
+
+  const ConnectionRef& conn = connect_to(type, addrs, false);
```

Comment:
i think it's simpler to just reply on the copy elision instead of storing the temporary value as a const reference.